docs/streams/developer-guide/processor-api.html

```diff
@@ -209,22 +209,18 @@ <h2><a class="toc-backref" href="#id1">Overview</a><a class="headerlink" href="#
 <span class="c1">// here, we create a `KeyValueStore&lt;String, Long&gt;` named &quot;persistent-counts&quot;.</span>
-<span class="kn">import</span> <span class="nn">org.apache.kafka.streams.processor.StateStoreSupplier</span><span class="o">;</span>
+<span class="kn">import</span> <span class="nn">org.apache.kafka.streams.state.StoreBuilder</span><span class="o">;</span>
 <span class="kn">import</span> <span class="nn">org.apache.kafka.streams.state.Stores</span><span class="o">;</span>
 
-<span class="c1">// Note: The `Stores` factory returns a supplier for the state store,</span>
-<span class="c1">// because that&#39;s what you typically need to pass as API parameter.</span>
-<span class="n">StateStoreSupplier</span> <span class="n">countStoreSupplier</span> <span class="o">=</span>
-  <span class="n">Stores</span><span class="o">.</span><span class="na">create</span><span class="o">(</span><span class="s">&quot;persistent-counts&quot;</span><span class="o">)</span>
-    <span class="o">.</span><span class="na">withKeys</span><span class="o">(</span><span class="n">Serdes</span><span class="o">.</span><span class="na">String</span><span class="o">())</span>
-    <span class="o">.</span><span class="na">withValues</span><span class="o">(</span><span class="n">Serdes</span><span class="o">.</span><span class="na">Long</span><span class="o">())</span>
-    <span class="o">.</span><span class="na">persistent</span><span class="o">()</span>
-    <span class="o">.</span><span class="na">build</span><span class="o">();</span>
+<span class="c1">// Using a `KeyValueStoreBuilder` to build a `KeyValueStore`.</span>
+<span class="n">StoreBuilder</span><span class="o">&lt;</span><span class="n">KeyValueStore</span><span class="o">&lt;</span><span class="n">String</span><span class="o">,</span> <span class="n">Long</span><span class="o">&gt;&gt;</span> <span class="n">countStoreSupplier</span> <span class="o">=</span>
+  <span class="n">Stores</span><span class="o">.</span><span class="na">keyValueStoreBuilder</span><span class="o">(</span>
+    <span class="n">Stores</span><span class="o">.</span><span class="na">persistentKeyValueStore</span><span class="o">(</span><span class="s">&quot;persistent-counts&quot;</span><span class="o">),</span>
+    <span class="n">Serdes</span><span class="o">.</span><span class="na">String</span><span class="o">(),</span>
+    <span class="n">Serdes</span><span class="o">.</span><span class="na">Long</span><span class="o">());</span>
+<span class="n">KeyValueStore</span><span class="o">&lt;</span><span class="n">String</span><span class="o">,</span> <span class="n">Long</span><span class="o">&gt;</span> <span class="n">countStore</span> <span class="o">=</span> <span class="n">countStoreSupplier</span><span class="o">.</span><span class="na">build</span><span class="o">();</span>
```

Comment:
The types don't align. Using `Stores` you get plain `byte[]`/`byte[]` stores:
```
        StoreBuilder<KeyValueStore<byte[], byte[]>> a = Stores.keyValueStoreBuilder(
            Stores.persistentKeyValueStore("name"),
            Serdes.ByteArray(),
            Serdes.ByteArray()
        );
```

Reply:
Yeah, but here I'm using the `Serdes.String()` and `Serdes.Long()` to create the StoreBuilder<KeyValueStore<String, Long>>, is there sth. wrong with this ?
```
StoreBuilder<KeyValueStore<String, Long>> countStoreSupplier = Stores.keyValueStoreBuilder(
  Stores.persistentKeyValueStore("persistent-counts"),
  Serdes.String(),
  Serdes.Long());
KeyValueStore<String, Long> countStore = countStoreSupplier.build();
```

Reply:
`Stores.persistentKeyValueStore("persistent-counts")` returns a `KeyValueBytesStoreSupplier` that extends `StoreSupplier<KeyValueStore<Bytes, byte[]>>` -- thus, the types are fixed.

If you want to have a String/Long store, you cannot use `Stores.persistentKeyValueStore("persistent-counts")` but need implement your own `StoreSupplier` with corresponding types.

I am fine either way -- but the example code should compile :) It think it's best if you code the example first and than c&p the code into the docs -- this ensures you get everything right.

Reply:
I think I have to revert my comment here... The example works as you describe it. Got confused... Sorry.

Reply:
np @mjsax in fact, actually I should have run it before putting into docs, that's true and it's just because I see we are doing like this everywhere ... xD
And for `KeyValueBytesStoreSupplier`, seems it always provides a `byte[]/byte[]` store for metering purpose in `InnerMeteredKeyValueStore` ? shall we also change `{@link org.apache.kafka.streams.processor.StateStoreSupplier}` [here](https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/state/internals/InnerMeteredKeyValueStore.java#L58) to `{@link org.apache.kafka.streams.state.StoreSupplier}` as the first one is deprecated ?

Reply:
@mjsax done.